src/reducers/users.tsx

```diff
@@ -0,0 +1,109 @@
+import invariant from 'invariant';
+import { Reducer } from 'redux';
+import { ActionType, createAction, getType } from 'typesafe-actions';
+
+type UserId = number;
+
+export type User = {
+  id: UserId;
+  name: string;
+  email?: string;
+  permissions?: Array<string>;
+};
+
+export const actions = {
+  loadCurrentUser: createAction('LOAD_CURRENT_USER', (resolve) => {
+    return (payload: { user: User }) => resolve(payload);
+  }),
+  loadUser: createAction('LOAD_USER', (resolve) => {
```

Comment:
In regards to my last comment, I can't think of why we would need this action so I suggest deleting it for now.